validation/utils_test.go

```diff
@@ -23,13 +24,27 @@ var isBigIntWithinBounds = func(actual *big.Int, bounds [2]*big.Int) bool {
 }
 
 // isWithinBounds returns true if actual is within bounds, where the bounds are [lower bound, upper bound] and are inclusive.
-var isWithinBounds = func(actual uint32, bounds [2]uint32) bool {
+func isWithinBounds[T uint32 | uint64](actual T, bounds [2]T) bool {
 	if bounds[1] < bounds[0] {
 		panic("bounds are in wrong order")
 	}
 	return (actual >= bounds[0] && actual <= bounds[1])
 }
 
+// assertInBounds fails the test (but not immediately) if the passed param is outside of the passed bounds.
+var assertInBounds = func(t *testing.T, name string, got *big.Int, want [2]*big.Int) {
```

Comment:
Consider changing this name `assertInBounds` --> `assertBigIntInBounds` to keep consistent format with `assertIntInBounds`